pkg/controller/acmechallenges/sync.go

```diff
@@ -69,7 +69,7 @@ func (c *controller) Sync(ctx context.Context, ch *cmacme.Challenge) (err error)
 	oldChal := ch
 	ch = ch.DeepCopy()
 
-	if ch.DeletionTimestamp != nil {
+	if !ch.DeletionTimestamp.IsZero() {
```

Comment:
Nice, I didn't know about this method.
Random thought: I see that it returns true if either the timestamp is nil or if [time is zero](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go#L63). I wonder if we can ever be in a state where the time is zero?

Reply:
Yeah, I wondered that too. I suppose only in unit-tests where you can set the value directly. 
In real-lift the API serve prevents clients from changing that field, I think.

Some documentation links:
 * https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion
 * https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers
 * https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#objectmeta-v1-meta

> DeletionTimestamp is RFC 3339 date and time at which this resource will be deleted. This field is set by the server when a graceful deletion is requested by the user, and is not directly settable by a client.